exporter/postgres/setup_pg12.sql

```diff
@@ -133,5 +133,141 @@ WHERE (ROUND(used/slots*100)::int) > p_percent;
 $function$;
 
 
+/*
+ * Table and functions for monitoring changes to pg_settings and pg_hba_file_rules system catalogs.
+ * Can't just do a raw check for the hash value since Prometheus only records numeric values for alerts
+ * Table allows recording of existing settings so it can be referred back to to see what changed
+ * If function returns 0, then NO settings have changed for either pg_settings or hba since last known valid state 
+ * If function returns 1, then just pg_settings have changed since last known valid state
+ * If function returns 2, then just hba has changed since last known valid state
+ * If function returns 3, then both pg_settings and hba have changed since last known valid state
+ */
+
+DROP TABLE IF EXISTS monitor.settings_checksum;
```

Comment:
hmm so we will loose history.

Reply:
True. But my other consideration here is that this setup script needs to be able to be run whenever we update pgmonitor to update all relevant objects that we need to use. So if we had to change the schema in the log table, it's hard to make that easy for users unless we just drop and recreate the table since we don't know what state/version their table was in before.

The only way around this that I know of is to make the objects we install to the DB an extension. But we also have different objects depends on the version of PG and there is no way to easily manage different versions of the same extension across different versions of PG.

Reply:
other option is based on alter table or create a new table import old data.

Reply:
Alter table statements will not work because we do not know the previous state their table may be in. Creating new one may work, but again, may still be tricky not being able to know the previous table's state.

Reply:
Just thinking about this more, I don't know if we should be encouraging the use of this table for actually storing change history. If people need a state history of their postgresql.conf and pg_hba.conf, they should be storing those in version control to capture not only the settings themselves but associated comments.

The purpose of this table is to provide an immediate reference for people to see what has changed outside of their normal configuration management. We can document that the history of this table is not guaranteed and only provided for said reference.

Reply:
Noting down ideas from discussion:
- Have the reset function also clear out the log table as well. Could have it truncate (to avoid bloat) and then just insert the current config as valid. That last part could be done by just calling the *_checksum() functions again, since if the table is empty it just inserts the current config as valid.

Reply:
Updated reset functions to clear history table.